Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/detect_features.h

```diff
@@ -0,0 +1,450 @@
+                                             1));
+
+    internal::PatchIdMapWrapper<PatchIdMap,
+            typename boost::property_traits<PatchIdMap>::value_type> wrapmap(patch_id_map, offset);
+    return connected_components(p, wrapmap, parameters::edge_is_constrained_map(eif)
+                                .face_index_map(fimap));
+
+}
+template <typename PolygonMesh, typename EdgeIsFeatureMap, typename PatchIdMap>
+typename boost::graph_traits<PolygonMesh>::faces_size_type
+detect_surface_patches(PolygonMesh& p,
+                       PatchIdMap& patch_id_map,
+                       EdgeIsFeatureMap& eif)
+{
+    return detect_surface_patches(p, patch_id_map, eif, parameters::all_default());
+}
+
+
+template<typename GT,
+         typename FT,
+         typename PolygonMesh,
+         typename EIFMap,
+         typename VNFEMap>
+ void sharp_call(PolygonMesh& pmesh,
+                 FT& angle_in_deg,
+                 EIFMap& edge_is_feature_map,
+                 VNFEMap& vnfe)
+{
+    // Initialize vertices
+    BOOST_FOREACH(typename boost::graph_traits<PolygonMesh>::vertex_descriptor vd, vertices(pmesh))
+    {
+        put(vnfe,vd, 0);
+    }
+    FT cos_angle ( std::cos(CGAL::to_double(angle_in_deg) * CGAL_PI / 180.) );
+
+    // Detect sharp edges
+    BOOST_FOREACH(typename boost::graph_traits<PolygonMesh>::edge_descriptor ed, edges(pmesh))
+    {
+        typename boost::graph_traits<PolygonMesh>::halfedge_descriptor he = halfedge(ed,pmesh);
+        if(is_border(he,pmesh) || angle_in_deg == FT() ||
+                (angle_in_deg != FT(180) && internal::is_sharp<PolygonMesh, GT>(pmesh,he,cos_angle))
+                )
+        {
+            put(edge_is_feature_map, edge(he, pmesh), true);
+            put(vnfe, target(he,pmesh), get(vnfe, target(he,pmesh))+1);
+            put(vnfe, source(he,pmesh), get(vnfe, source(he,pmesh))+1);
+
+        }
+    }
+}
+
+
+template<typename GT,
+         typename FT,
+         typename PolygonMesh,
+         typename EIFMap>
+ void sharp_call(PolygonMesh& pmesh,
+                 FT& angle_in_deg,
+                 EIFMap& edge_is_feature_map,
+                 const boost::param_not_found&)
+{
+    FT cos_angle ( std::cos(CGAL::to_double(angle_in_deg) * CGAL_PI / 180.) );
+
+    // Detect sharp edges
+    BOOST_FOREACH(typename boost::graph_traits<PolygonMesh>::edge_descriptor ed, edges(pmesh))
+    {
+        typename boost::graph_traits<PolygonMesh>::halfedge_descriptor he = halfedge(ed,pmesh);
+        if(is_border(he,pmesh) || angle_in_deg == FT() ||
+                (angle_in_deg != FT(180) && internal::is_sharp<PolygonMesh, GT>(pmesh,he,cos_angle))
+                )
+        {
+            put(edge_is_feature_map, edge(he, pmesh), true);
+        }
+    }
+}
+} //end internal
+
+/*!
+ * \ingroup PMP_detect_features_grp
+ *
+ * detects and marks as constrained edges around which incident faces form an angle equal to or greater than `angle_in_deg`.
```

Comment:
I still disagree with that definition.
See [detect_features() in Mesh_3](http://doc.cgal.org/latest/Mesh_3/classCGAL_1_1Polyhedral__mesh__domain__with__features__3.html#a5a868ac7b8673436766d28b7a80d2826) for a correct definition. Commit 99aec9eda9647730db2a5016ee4be0dfce29b65e will fix it.